dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/OracleEvent.scala

```diff
@@ -0,0 +1,268 @@
+package org.bitcoins.dlc.oracle
+
+import java.time.Instant
+
+import org.bitcoins.commons.jsonmodels.dlc.SigningVersion
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.tlv._
+import org.bitcoins.crypto._
+import org.bitcoins.dlc.oracle.storage.EventDb
+
+sealed trait OracleEvent {
+
+  /** The nonces the oracle is committing to for this event */
+  def nonces: Vector[SchnorrNonce]
+
+  /** The oracle's public key */
+  def pubkey: SchnorrPublicKey
+
+  /** The name given to this event, may be a URI */
+  def eventName: String
+
+  /** The version of signing for this event */
+  def signingVersion: SigningVersion
+
+  /** The earliest expected time an outcome will be signed */
+  def maturationTime: Instant
+
+  /** A signature by the oracle of the hash of nonce and event name */
+  def announcementSignature: SchnorrDigitalSignature
+
+  def eventDescriptorTLV: EventDescriptorTLV
+
+  def eventTLV: OracleEventTLV =
+    OracleEventV0TLV(pubkey,
+                     UInt32(maturationTime.getEpochSecond),
+                     eventDescriptorTLV,
+                     eventName)
+
+  def announcementTLV: OracleAnnouncementTLV = {
+    eventTLV match {
+      case v0TLV: OracleEventV0TLV =>
+        OracleAnnouncementV0TLV(announcementSignature, v0TLV)
+    }
+  }
+}
+
+sealed trait PendingOracleEvent extends OracleEvent
+
+sealed trait CompletedOracleEvent extends OracleEvent {
```

Comment:
can't we add an invariant here actually checking to see the signatures are valid?

Reply:
We could but it'd be very inefficient. We'd need to go through and check against each outcome